### PR TITLE
Refactor SaveAction

### PR DESCRIPTION
**Pull request by koppor**
While working on https://github.com/JabRef/jabref/issues/6109, we found, we need to refactor the `save` things to understand what's going on. ^^

- Get rid of "doSave()"
- Add notification for non-successful save in case of "SaveAll"
- Remove deprecated "setDatabaseFile" (as we are Java8 with Path)

### CODE REVIEW

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java [hunk 1]
```diff
@@ -161,32 +163,36 @@ private boolean doSave() {
         }
     }
 
-    public boolean save() {
-        return save(SaveDatabaseMode.NORMAL);
+    public boolean save(BibDatabaseContext bibDatabaseContext) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java]: Why is the database context passed as an method argument? The base panel is already an constructor argument, thus I would expect that the `SaveAction` always operates on this database. Maybe set the database context in a new field in the constructor?

**Comment by stefan-kolb** [comment]
@koppor I think we had a bug with this when switching tabs and saving. We should thoroughly test this before merging!

**Comment by calixtus** [comment]
Just a spontaneous thought:
Could the SaveDatabaseAction class be devided into a logic part (which should reside in the logic package) and a gui part, which could be merged with the SaveAction class?

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java [hunk 2]
```diff
@@ -141,140 +195,88 @@ public boolean save(Path targetPath, SaveDatabaseMode mode) {
...
-            context.setDatabasePath(oldFile);
-            AutosaveManager.shutdown(context);
-            BackupManager.shutdown(context);
+            if (fileWriter.hasEncodingProblems()) {
+                saveWithDifferentEncoding(file, selectedOnly, preferences.getEncoding(), fileWriter.getEncodingProblems(), saveType);
+            }
+        } catch (UnsupportedCharsetException ex) {
+            throw new SaveException(Localization.lang("Character encoding '%0' is not supported.", encoding.displayName()), ex);
+        } catch (IOException ex) {
+            throw new SaveException("Problems saving:", ex);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java]: can we add here a logger instead?
The Problems saving often misses the ex

**Comment by koppor** [comment]
A log of the file that is saved would be good. Currently, we only have following log output:

    22:26:04.530 [JavaFX Application Thread] INFO  org.jabref.gui.JabRefDialogService - Saving library...

The only issue remaining is that with "Save As", the tab title is not changed if the filename changed - it works in the `master` branch. - No clue, where this happened. As a quick hack, we could call `panel.markChangedOrUnChanged()`.

**Comment by koppor** [comment]
No issue when switching tabs - we might have introduce that in another PR.

**Comment by calixtus** [comment]
I think calling panel.markChangedOrUnChanged is fine at the moment. You probably should comment it as FIXME. This should all then be done when BibDatabaseContext, BasePanel and JabRefFrame are untangeld - and the UndoManager is fixed.

**Pull request merged**